ambry-replication/src/main/java/com.github.ambry.replication/PartitionInfo.java

```diff
@@ -50,6 +53,44 @@ public ReplicaId getLocalReplicaId() {
     return this.localReplicaId;
   }
 
+  /**
+   * Add {@link RemoteReplicaInfo} to this {@link PartitionInfo} if it is previously absent.
+   * @param remoteReplicaInfo the {@link RemoteReplicaInfo} to add.
+   * @return {@code true} if remote replica info is added. {@code false} if it is already present
+   */
+  boolean addReplicaInfoIfAbsent(RemoteReplicaInfo remoteReplicaInfo) {
+    lock.lock();
```

Comment:
please lock getRemoveReplicaInfos method as well, this lock is protecting remoteReplicas list, we should lock it everywhere we use it.

Reply:
make sense, i will use read-write lock instead